tests/test-extensions/test-config-properties-extension/src/main/java/org/mule/test/crafted/config/properties/extension/SecureConfigurationPropertiesProvider.java

```diff
@@ -34,11 +44,44 @@ public SecureConfigurationPropertiesProvider(ResourceProvider resourceProvider,
     this.mode = mode;
   }
 
+  @Override
+  public void initialise() throws InitialisationException {
+    super.initialise();
+    initializationCount++;
+  }
+
+  @Override
+  public void dispose() {
+    disposeCount++;
+  }
+
   @Override
   public Optional<ConfigurationProperty> getConfigurationProperty(String configurationAttributeKey) {
     if (configurationAttributeKey.startsWith(SECURE_PREFIX)) {
       String effectiveKey = configurationAttributeKey.substring(SECURE_PREFIX.length());
       return Optional.ofNullable(configurationAttributes.get(effectiveKey));
+    } else if (configurationAttributeKey.startsWith(LIFECYCLE_PREFIX)) {
```

Comment:
what's the purpose of this?

Reply:
It't the only solution i could come up with to test that the lifecycle phases were applied. It was either that or declaring a public static field  or getter accesible from the test class.
Keep in mind that the configuration properties extension is just used for testing purposes.